exchange/exchange.go

```diff
@@ -107,6 +109,15 @@ type AuctionRequest struct {
 	LegacyLabels pbsmetrics.Labels
 }
 
+// BidderRequest holds the bidder specific request and all other
+// information needed to process that bidder request.
+type BidderRequest struct {
```

Comment:
I went back and forth on this name a few times. I don't really like the redundancy in the names here such as `BidderRequest.BidRequest`, `BidderRequest.BidderName` and such but the name `Bidder` doesn't convey much information and the other name that I played with was `AuctionRequestBidder` but didn't like that much either so went back to `BidderRequest`. Happy to consider any other suggestions if there are.

Reply:
I like the `BidderRequest` request. You could remove the Bid/Bidder prefixes on the fields to reduce the naming redundancy, like:

```
type BidderRequest struct {
	Request     *openrtb.BidRequest
	Name     openrtb_ext.BidderName
	CoreName openrtb_ext.BidderName
	Labels   pbsmetrics.AdapterLabels
}
```

Reply:
I considered this idea but the issue with that though is then the field names don't really make sense based on the object name i.e. the field `Name` or `CoreName` doesn't belong to a bidder _request_. It belongs to a _bidder_. And using just `Name` and `CoreName` with `BidderRequest` as the object name doesn't convey that. This is where I had moved towards using just `Bidder` for the object name but then that didn't make enough sense for this object name either

Reply:
Gotcha. Let's just fo with this then. It's an internal data structure which we can always rename again in the future if we think of something better.